packages/@lwc/module-resolver/src/index.ts

```diff
@@ -202,8 +202,8 @@ export function resolveModule(
             `The importee argument must be a valid LWC module name. Received "${importee}"`
         );
     }
-
-    const rootDir = findFirstUpwardConfigPath(path.resolve(dirname));
+    const customRootDir = config && config.rootDir ? config.rootDir : null;
+    const rootDir = findFirstUpwardConfigPath(path.resolve(customRootDir || dirname));
```

Comment:
I think this matches the old functionality of `rootDir`: https://github.com/salesforce/lwc/blob/ce73c8e906fc9aba884b520d87b6191332dc4f11/packages/%40lwc/module-resolver/src/index.ts#L117